src/main/java/uk/gov/gchq/palisade/rule/Rules.java

```diff
@@ -175,35 +132,8 @@ public boolean containsRules() {
 
     @Override
     public boolean equals(final Object o) {
-        boolean rtn = (this == o);
```

Comment:
I dont think this equals will work
something like this will be better:

```
@Override
    public boolean equals(final Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof Rules)) {
            return false;
        }
        Rules<?> other = (Rules<?>) o;
        LinkedList<Class<? extends Rule>> thisRuleClasses = this.rulesMap.values().stream()
                .map(Rule::getClass)
                .collect(Collectors.toCollection(LinkedList::new));
        LinkedList<Class<? extends Rule>> otherRuleClasses = other.rulesMap.values().stream()
                .map(Rule::getClass)
                .collect(Collectors.toCollection(LinkedList::new));
        return thisRuleClasses.equals(otherRuleClasses);
    }
```

Reply:
Done, with null handling